dev_guide/getting_traffic_into_cluster.adoc

```diff
@@ -133,10 +146,298 @@ xref:../admin_guide/high_availability.adoc#admin-guide-high-availability[VIP] ad
+
+After instantiating the template, a ClusterIP-based service and
+`DeploymentConfig` is created and a new pod containing PostgreSQL will be
+started.
+
+[[configuring-an-IP-address-for-a-service]]
+==== Configuring an IP Address for a Service
+
+To allow the cluster to automatically assign an IP address for a service, create
+a service definition similar to the following that will create a new Ingress
+service:
+
+----
+$ oc create -f - <<INGRESS
+apiVersion: v1
+kind: Service
+metadata:
+  name: postgresql-ingress
+spec:
+  ports:
+  - name: postgresql
+    port: 5432
+  type: LoadBalancer <1>
+  selector:
+    name: postgresql
+INGRESS
+----
+<1> The `LoadBalancer` type of service will make the request for an external service
+on behalf of the application user.
+
+Alternatively, the `oc expose` command can be used to create the service:
+
+----
+$ oc expose dc postgresql --type=LoadBalancer --name=postgresql-ingress
+----
+
+Once the service is created, the external IP address is automatically allocated
+by the cluster and can be confirmed by running:
+
+----
+$ oc get svc postgresql-ingress
+----
+
+.Example oc get Output
+----
+NAME         CLUSTER-IP      EXTERNAL-IP   PORT(S)   AGE
+postgresql-ingress    172.30.74.106   10.9.54.100,10.9.54.100    5432/TCP    30s
+----
+
+Specifying the type `LoadBalancer` also configures the service with a
+`nodePort` value, allowing direct access to the service on each of the nodes the
+application is deployed on. To discover the node port automatically assigned,
+run:
+
+----
+$ oc export svc postgresql-ingress
+----
+
+.Example oc export Output
+----
+apiVersion: v1
+kind: Service
+metadata:
+  creationTimestamp: null
+  labels:
+    app: postgresql-persistent
+    template: postgresql-persistent-template
+  name: postgresql-ingress
+spec:
+  deprecatedPublicIPs:
+  - 10.9.54.100
+  externalIPs:
+  - 10.9.54.100
+  ports:
+  - nodePort: 32439 <1>
+    port: 5432
+    protocol: TCP
+    targetPort: 5432
+  selector:
+    name: postgresql
+  sessionAffinity: None
+  type: LoadBalancer
+----
+<1> Automatically assigned port.
+
+A PostgreSQL client can now be configured to connect directly to any node using
+the value of the assigned `nodePort`.
+
+[[configuring-the-service-to-be-highly-available]]
+==== Configuring the Service to be Highly Available
+
+Instead of connecting directly to individual nodes, you can use one of
+{product-title}'s
+xref:../admin_guide/high_availability.adoc#admin-guide-high-availability[high
+availability] strategies by deploying the IP failover router to provide access
+services configured with external IP addresses. This allows cluster
+administrators the flexibility of defining the ingress points within a cluster,
+and ensuring the service is readily available.
+
+[NOTE]
+====
+Nodes that have IP failover routers deployed to them must be in the same *Layer
+2* switching domain for ARP broadcasts to communicate to switches what
+appropriate port the destination should flow to.
```

Comment:
"switch" port. 

I am not sure where this is going. What does it take to have a working cluster that breaks this?

Stray comment: each port on the switch is wired to a NIC on one of the nodes. The switch keeps track of which mac addresses arrive on each switch port so when a packet arrives on some port it can be directly "switched" the port that saw its destination mac. Hence the device is called a switch. Switches are layer 2 and arp broadcasts are layer 2 so an arp broadcast goes to all ports on the layer 2 segment. I think that all of the nodes are in the same layer 2 segment (Ben ?)